crates/solver/src/settlement.rs

```diff
@@ -278,9 +289,25 @@ impl Settlement {
         Ok(Self { encoder: merged })
     }
 
-    // Checks if the settlement is safe from MEV extraction
-    pub fn mev_safe(&self) -> bool {
-        true
+    // Calculates the risk level for settlement to be reverted
+    pub fn revertable(&self) -> Revertable {
+        if self.encoder.interactions().is_empty() {
+            return Revertable::NoRisk;
+        }
+
+        match self
+            .encoder
+            .interactions()
+            .iter()
+            .filter_map(|interaction| {
+                let (token, amount) = interaction.executed_amount()?;
+                amount.checked_mul(self.clearing_price(token)?)
+            })
+            .any(|executed_amount| executed_amount > U256::from_f64_lossy(5e36))
```

Comment:
5e36 is bad way I guess, since not all tokens have 18digits base right?

Reply:
Aren't clearing prices already computed in WETH so it wouldn't matter what precision each token has?

Reply:
Hm, clearing price for ETH/WETH is "0x0de0b6b3a7640000" = 1000000000000000000, so they are denominated in ETH but expressed in wei.

Reply:
The same is true for amount, so basically need to bump x18 twice